### ISSUE TITLE
prefer-const wrongly reports `let` declarations (without initialization)

### ISSUE THREAD

**Issue by Cortys** [eslint/eslint#5304]
Using eslint 2.1.0 (also reproducible in 2.0.0).

**Example:**
```js
"use strict";

let a; // ESLint error: prefer-const: 'a' is never modified, use 'const' instead.

if(Math.random() > 0.5)
	a = 1;

console.log(a);
```

`let` declarations without initialization are wrongly rejected by the `prefer-const` rule, which is nonsensical since there are no `const` declarations without init value is JS.

Restructuring the code to use `const` would only be possible by using a ternary expression (e. g. `const a = Math.random() > 0.5 ? 1 : undefined`) and I do not think that `prefer-const` should enforce this.

**Config:** To reproduce this issue, enabling the `prefer-const` rule should be sufficient.

**Comment by gyandeeps** [comment]
Duplicate: https://github.com/eslint/eslint/issues/5283

**Comment by Cortys** [comment]
I see. Didn't see those since they are closed.

I guess this can be closed as well in favor of #5284.

**Comment by alberto** [comment]
I'm not sure it's the exact same issue, since in this case `a` is used outside the `if` consequent. @mysticatea ?

**Comment by mysticatea** [comment]
This `if` block does not have a block.
I must consider this pattern in #5284.
